event_model/__init__.py

```diff
@@ -1797,353 +1872,647 @@ def __iter__(self):
         )
 
 
-ComposeDescriptorBundle = namedtuple(
-    "ComposeDescriptorBundle", "descriptor_doc compose_event compose_event_page"
-)
-ComposeResourceBundle = namedtuple(
-    "ComposeResourceBundle", "resource_doc compose_datum compose_datum_page"
-)
-ComposeStreamResourceBundle = namedtuple(
-    "ComposeStreamResourceBundle", "stream_resource_doc compose_stream_data"
-)
+@dataclass
+class ComposeResourceBundle:
+    resource_doc: Resource
+    compose_datum: Callable
+    compose_datum_page: Callable
```

Comment:
```suggestion
    compose_datum: ComposeDatum
    compose_datum_page: ComposeDatumPage
```
Also, could you move this class to just above Compose Resource please?